### DISCUSSION TITLE
argo-workflows/http-template/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9714] [General]
# argo-workflows/http-template/

https://argoproj.github.io/argo-workflows/http-template/

**Comment by gocpplua** [comment]
Events:
```
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    7m21s                 default-scheduler  Successfully assigned argo/http-template-qxg5w-1340600742-agent to cn0314000510l
  Warning  FailedMount  69s (x11 over 7m21s)  kubelet            MountVolume.SetUp failed for volume "kube-api-access-cj9fl" : secret "default.service-account-token" not found
  Warning  FailedMount  45s (x3 over 5m18s)   kubelet            Unable to attach or mount volumes: unmounted volumes=[kube-api-access-cj9fl], unattached volumes=[var-run-argo kube-api-access-cj9fl]: timed out waiting for the condition
```

**Comment by jiangshihao01** [comment]
Why does each http request take more than 30 seconds to complete, even if the url I requested is immediately returned, such as www.google.com

**Reply by agilgur5** [reply]
I imagine the request itself completes immediately, but it takes time for a Workflow to be queued, a Pod to be created, scheduled  and run, and then the results to be reported back.
If you have a lot of Workflows running, then you could have a longer queue. And if you're using synchronization, then you have to wait for locks etc to be released. Plus k8s itself has to schedule and create the Pod; the latency there depends on your cluster availability, image cache, etc.

The UI should show how much time the Pod actually spent running

**Comment by Bryce-huang** [comment]
How to configure the agent's image registry?
